internal/handlers/categoryHandler.go

```diff
@@ -53,6 +56,60 @@ func GetCategories(c *fiber.Ctx) error {
 		}
 		categories = append(categories, category)
 	}
-
 	return c.Render("category", categories)
 }
+
+// GetCategory method retrieves a category from the categories table
+func GetCategory(p providers.CategoryProvider) func(c *fiber.Ctx) error {
+	return func(c *fiber.Ctx) error {
+		id := c.Params("id")
+		categoryID, _ := strconv.Atoi(id)
+		category, err := p.CategoryGet(categoryID)
+
+		if err != nil {
+			return err
+		}
+
+		// wrapped into array to it works in the template
+		result := make([]*models.Category, 0)
+		result = append(result, category)
+		return c.Render("category", result)
+	}
+}
+
+// UpdateCategory method update a category from the categories table and returns an error
+func UpdateCategory(p providers.CategoryProvider) func(c *fiber.Ctx) error {
+	return func(c *fiber.Ctx) error {
+		id := c.Params("id")
+		categoryID, _ := strconv.Atoi(id)
```

Comment:
this line could be lower: cat.ID, _ := strconv.Atoi(id) - then you wouldn't need a second ID parameter. But the way you wrote it is ok.